src/target/riscv/riscv-013.c

```diff
@@ -1741,6 +1744,18 @@ static int examine(struct target *target)
 		r->mtopi_readable = false;
 	}
 
+	riscv_reg_t dcsr;
```

Comment:
Does this code get reached if power down/up happens after the first call to "init" command happens?

Reply:
Yes, but the mechanism is fragile. What happens is you get a failure in poll(), and then in handle_target() that failure is seen and it calls target_reset_examined().

I'll try to make this mechanism more robust.